### PR TITLE
Series block sort order

### PR DESCRIPTION
**Pull request by JulianKniephoff**
The frontend now sorts the videos in a series block according to the setting.

Note that this includes #377 for no particular reason other than my own convenience. :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: frontend/src/util/index.ts [hunk 1]
```diff
@@ -56,6 +56,19 @@ export function keyOfId(id: string): string {
...
 
+/**
+ * Create a comparison function for `Array.prototype.sort` comparing whatever
+ * the given key function returns as numbers.
+ */
+export function compareBy<T>(key: (item: T) => number): (itemA: T, itemB: T) => number {
+    return (itemA, itemB) => key(itemB) - key(itemA);
+}
+
+/** Swap a binary function's arguments. Useful for example for comparison funcitons. */
```
Comment (LukasKalbertodt) [inline frontend/src/util/index.ts:67]: `funcitons` typo

**Review by LukasKalbertodt — CHANGES_REQUESTED** [review]
Seems to work. Just a few comments about code.

File: frontend/src/ui/Blocks/Series.tsx [hunk 2]
```diff
@@ -85,8 +86,22 @@ export const SeriesBlock: React.FC<Props> = ({
...
+    switch (order) {
+        case "NEW_TO_OLD":
+            sortedEvents.sort(compareNewToOld);
+            break;
+        case "OLD_TO_NEW":
+            sortedEvents.sort(compareOldToNew);
+            break;
+        default:
+            return unreachable();
+    }
```
[the remark this answers is not in the document]
Comment (JulianKniephoff) [reply frontend/src/ui/Blocks/Series.tsx:102]: That would force us to handle the Relay-internal `"%future added value"`:

```
TS2345: Argument of type '{ NEW_TO_OLD: () => { readonly id: string; readonly title: string; readonly thumbnail: string | null; readonly duration: number; readonly created: string; readonly creators: readonly string[]; readonly tracks: readonly { ...; }[]; }[]; OLD_TO_NEW: () => { readonly id: string; readonly title: string; readonly thumbna...' is not assignable to parameter of type 'Record<VideoListOrder, () => { readonly id: string; readonly title: string; readonly thumbnail: string | null; readonly duration: number; readonly created: string; readonly creators: readonly string[]; readonly tracks: readonly { ...; }[]; }[]>'.
  Property '"%future added value"' is missing in type '{ NEW_TO_OLD: () => { readonly id: string; readonly title: string; readonly thumbnail: string | null; readonly duration: number; readonly created: string; readonly creators: readonly string[]; readonly tracks: readonly { ...; }[]; }[]; OLD_TO_NEW: () => { readonly id: string; readonly title: string; readonly thumbna...' but required in type 'Record<VideoListOrder, () => { readonly id: string; readonly title: string; readonly thumbnail: string | null; readonly duration: number; readonly created: string; readonly creators: readonly string[]; readonly tracks: readonly { ...; }[]; }[]>'.
```

[the remark this answers is not in the document]
Comment (LukasKalbertodt) [reply frontend/src/ui/Blocks/Series.tsx:102]: `match` has an optional extra argument for "default", i.e. what if no arm matches.

[the remark this answers is not in the document]
Comment (JulianKniephoff) [reply frontend/src/ui/Blocks/Series.tsx:102]: Yes. I am not very smart sometimes. :|

**Pull request merged**